pragma-node/src/infra/repositories/onchain_repository.rs

```
@@ -131,6 +134,114 @@ fn build_sql_query(
+        .await?;
+        return Ok((price, sources, vec![pair_id], None));
+    }
+
+    let offchain_conn = offchain_pool.get().await.map_err(adapt_infra_error)?;
+
+    let alternative_currencies = offchain_conn
+        .interact(Currency::get_abstract_all)
+        .await
+        .map_err(adapt_infra_error)?
+        .map_err(adapt_infra_error)?;
+    let (base, quote) = pair_id.split_once('/').unwrap_or(("", ""));
+
+    for alt_currency in alternative_currencies {
+        let base_alt_pair = format!("{}/{}", base, alt_currency);
+        let alt_quote_pair = format!("{}/{}", quote, alt_currency);
+
+        if onchain_pair_exist(&existing_pair_list, &base_alt_pair)
+            && onchain_pair_exist(&existing_pair_list, &alt_quote_pair)
+        {
+            let mut base_alt_result = get_sources_and_aggregate(
+                onchain_pool,
+                network,
+                base_alt_pair.clone(),
+                timestamp,
+                aggregation_mode,
+            )
+            .await?;
+            let base_alt_decimal = get_decimals(offchain_pool, &base_alt_pair).await?;
+            let alt_quote_result = get_sources_and_aggregate(
+                onchain_pool,
+                network,
+                alt_quote_pair.clone(),
+                timestamp,
+                aggregation_mode,
+            )
+            .await?;
+            let alt_quote_decimal = get_decimals(offchain_pool, &alt_quote_pair).await?;
+
+            let rebased_price = calculate_rebased_price(
+                (base_alt_result.0, base_alt_decimal),
+                (alt_quote_result.0, alt_quote_decimal),
+            )?;
+            base_alt_result.1.extend(alt_quote_result.1);
+            return Ok((
+                rebased_price.0,
+                base_alt_result.1,
+                vec![base_alt_pair, alt_quote_pair],
+                Some(rebased_price.1),
+            ));
+        }
+    }
+    Ok((BigDecimal::from(0), vec![], vec![], None))
```

This should return a 404 error